pages/k8s/1.29/upgrading.md

```diff
@@ -0,0 +1,396 @@
+  <div markdown="1" class="p-notification__content">
+    <span class="p-notification__title">Note:</span>
+    <p class="p-notification__message"><strong>The 'bionic' series (Ubuntu 18.04) is no longer supported:</strong> Support for bionic series charms in relation to Charmed Kubernetes has expired. If you are running charms on 'bionic', you will need to series upgrade the charms before completing the rest of the upgrade procedure.</p>
+  </div>
+</div>
+
+
+## Upgrading the Machine's Series (required for machines currently running 18.04(Bionic))
+
+All of the charms support [upgrading the machine's series via Juju](https://juju.is/docs/juju/manage-machines#heading--upgrade-a-machine).
+As each machine is upgraded, the applications on that machine will be stopped and the unit will
+go into a `blocked` status until the upgrade is complete. For the worker units, pods will be drained
+from the node and onto one of the other nodes at the start of the upgrade, and the node will be removed
+from the pool until the upgrade is complete.
+
+
+## Infrastructure updates
+
+The applications which run alongside the core Kubernetes components can be upgraded
+at any time. These applications are widely used and may frequently receive upgrades
+outside of the cycle of new releases of Kubernetes.
+
+This includes:
+
+- easyrsa
+- etcd
+- flannel, calico or other CNI
+
+Note that this may include other applications which you may have installed, such as
+Elasticsearch, Prometheus, Nagios, Helm, etc.
+
+
+<a id='upgrading-containerd'> </a>
+
+### Upgrading Containerd
+
+By default, Versions 1.15 and later use Containerd as the container
+runtime. This subordinate charm can be upgraded with the command:
+
+```bash
+juju refresh containerd --channel=1.29/stable
+```
+
+### Upgrading etcd
+
+As **etcd** manages critical data for the cluster, it is advisable to create a snapshot of
+this data before running an upgrade. This is covered in more detail in the
+[documentation on backups][backups] (including how to restore snapshots in case of
+problems).
+
+Upgrade the charm with the command:
+
+```bash
+juju refresh etcd --channel=1.29/stable
+```
+
+To upgrade **etcd** itself, you will need to set the **etcd** charm's channel
+config.
+
+To determine the correct channel, go to the
+[releases section of the bundle repository][bundle-repo] page and check the relevant
+**Charmed Kubernetes** bundle. Within the bundle, you should see which channel
+the **etcd** charm is configured to use.
+
+Once you know the correct channel, set the **etcd** charm's channel config:
+
+```bash
+juju config etcd channel=3.4/stable
+```
+### Upgrading MetalLB (if used)
+
+Previous versions of the Charmed Kubernetes bundle adopted a two charm approach, deployed in a K8s model with the suggested name `metallb-system`.
+The 1.28 release includes the charms for MetallLB have been updated into a single charm (`metallb`). Updating is simply a matter of removing the old
```

Comment:
```suggestion
The 1.29 release includes the charms for MetallLB have been updated into a single charm (`metallb`). Updating is simply a matter of removing the old
```

Reply:
i worded it a bit differently.  1.28 was the first release that included the new style metallb and i wanted to keep that tidbit known to folks.  i.e. if they want to go from 1.27 -> 1.28 -> 1.29, hopefully they'll read this and know the architecture changed at 1.28.